shell/platform/fuchsia/flutter/pointer_injector_delegate.cc

```diff
@@ -0,0 +1,318 @@
+    auto args_it = request.FindMember("args");
+    if (args_it == request.MemberEnd() || !args_it->value.IsObject()) {
+      FML_LOG(ERROR) << "No arguments found.";
+      return false;
+    }
+
+    const auto& args = args_it->value;
+
+    auto view_id = args.FindMember("viewId");
+    if (!view_id->value.IsUint64()) {
+      FML_LOG(ERROR) << "Argument 'viewId' is not a uint64";
+      return false;
+    }
+    auto id = view_id->value.GetUint64();
+
+    auto phase = args.FindMember("phase");
+    if (!phase->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'phase' is not a int";
+      return false;
+    }
+
+    auto pointer_x = args.FindMember("x");
+    if (!pointer_x->value.IsFloat() && !pointer_x->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'Pointer.X' is not a float";
+      return false;
+    }
+
+    auto pointer_y = args.FindMember("y");
+    if (!pointer_y->value.IsFloat() && !pointer_y->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'Pointer.Y' is not a float";
+      return false;
+    }
+
+    auto pointer_id = args.FindMember("pointerId");
+    if (!pointer_id->value.IsUint()) {
+      FML_LOG(ERROR) << "Argument 'pointerId' is not a uint32";
+      return false;
+    }
+
+    auto trace_flow_id = args.FindMember("traceFlowId");
+    if (!trace_flow_id->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'traceFlowId' is not a int";
+      return false;
+    }
+
+    auto view_ref = args.FindMember("viewRef");
+    if (!view_ref->value.IsUint64()) {
+      FML_LOG(ERROR) << "Argument 'viewRef' is not a uint64";
+      return false;
+    }
+
+    zx_handle_t handle = view_ref->value.GetUint64();
+    zx_handle_t out_handle;
+    zx_status_t status =
+        zx_handle_duplicate(handle, ZX_RIGHT_SAME_RIGHTS, &out_handle);
+    if (status != ZX_OK) {
+      FML_LOG(ERROR) << "Argument 'viewRef' is not valid";
+      return false;
+    }
+    auto ref = fuv_ViewRef({
+        .reference = zx::eventpair(out_handle),
+    });
+
+    auto width = args.FindMember("logicalWidth");
+    if (!width->value.IsFloat() && !width->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'logicalWidth' is not a float";
+      return false;
+    }
+
+    auto height = args.FindMember("logicalHeight");
+    if (!height->value.IsFloat() && !height->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'logicalHeight' is not a float";
+      return false;
+    }
+
+    auto timestamp = args.FindMember("timestamp");
+    if (!timestamp->value.IsInt()) {
+      FML_LOG(ERROR) << "Argument 'timestamp' is not a int";
+      return false;
+    }
+
+    PointerInjectorRequest request = {
+        .x = pointer_x->value.GetFloat(),
+        .y = pointer_y->value.GetFloat(),
+        .pointer_id = pointer_id->value.GetUint(),
+        .phase = static_cast<fup_EventPhase>(phase->value.GetInt()),
+        .trace_flow_id = trace_flow_id->value.GetUint64(),
+        .view_ref = std::move(ref),
```

Comment:
This should come from the view_id->view_ref map

Reply:
No need. We can use OnChildViewViewRef for this